### PR TITLE
-Added Get Mapping for deleting user

### PR DESCRIPTION
**Pull request by JoMuelle**
Mapping For deleteUser requests (currently no response, as in RestInterface, should I change that?)

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 1]
```diff
@@ -157,4 +157,12 @@ public boolean isAvailable(User userInput) {
     User user = userRepository.findByUsername(userInput.getUsername());
     return user == null;
   }
+
+  public void deleteUser(long userId){
+      User userToDelete = this.userRepository.findById(userId);
+      if(userToDelete == null){
+          throw new ResponseStatusException(HttpStatus.NOT_FOUND, "user does not exit"); //Throws 404 if user with userId doesn't exist
+      }
+      userRepository.deleteById(userId);
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:166]: What happens if you just delete by id where the user doesn't exist (without getting the user and checking if it's null)? 
Do you have to flush the repo or is this done in deleteById?

**Review by daveking7 — COMMENTED** [review]
See comments

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java (hunk 1 above)
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:166]: Not sure, will try to find out

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 2]
```diff
@@ -157,4 +157,12 @@ public boolean isAvailable(User userInput) {
     User user = userRepository.findByUsername(userInput.getUsername());
     return user == null;
   }
+
+  public void deleteUser(long userId){
+      User userToDelete = this.userRepository.findById(userId);
+      if(userToDelete == null){
```
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:163]: Ahh yes, makes sense i think (will go over it, probably will delete these lines)

[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:163]: Fixed now

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 3]
```diff
@@ -157,4 +157,12 @@ public boolean isAvailable(User userInput) {
     User user = userRepository.findByUsername(userInput.getUsername());
     return user == null;
   }
+
+  public void deleteUser(long userId){
+      User userToDelete = this.userRepository.findById(userId);
```
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:162]: redundant, as lines deleted (checked by specific access, see above)

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java (hunk 1 above)
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:166]: should be done

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 4]
```diff
@@ -110,7 +110,7 @@ private void checkIfUserExistsForNewUsername(User user) {
      * and throw an error otherwise.
      *
      * @param inputUser user as input
-     * @throws org.springframework.web.server.ResponseStatusException: 401
+     * @throws org.springframework.web.server.ResponseStatusException
```
Comment (serainaschraff) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:113]: why did you change that? it does throw a 401 doesn't it?

**Review by serainaschraff — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java (hunk 4 above)
Comment (JoMuelle) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:113]: Probably was on accident, didnt want to change it
